### PR TITLE
Add style performance statistics to Servo

### PR DESCRIPTION
**Pull request by bholley**
Gecko bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1331856

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15119)

### CODE REVIEW

File: components/style/parallel.rs [hunk 1]
```diff
@@ -78,13 +73,16 @@ pub fn traverse_dom<E, D>(traversal: &D,
...
-        let misses = STYLE_SHARING_CACHE_MISSES.load(Ordering::SeqCst);
-
-        println!("Style sharing stats:");
-        println!(" * Hits: {}", hits);
-        println!(" * Misses: {}", misses);
+    // Dump statistics to stdout if requested.
+    if TraversalStatistics::should_dump() || opts::get().style_sharing_stats {
+        let slots = unsafe { tls.unsafe_get() };
+        let aggregate = slots.iter().fold(TraversalStatistics::default(), |acc, t| {
+            match *(t.borrow()) {
```
Comment (emilio) [inline components/style/parallel.rs]: Forgot to fix this in the Servo patch? Not a big deal though.

Comment (bholley) [inline components/style/parallel.rs]: Sorry, it was in my local tree but I didn't |git commit -a --amend|. I'll do it now.

**Pull request merged**